<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>590</width>
    <height>405</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>&amp;Database</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::ExpandingFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Database &amp;encoding</string>
         </property>
         <property name="buddy">
          <cstring>encodingComboBox</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="encodingComboBox">
         <item>
          <property name="text">
           <string notr="true">UTF-8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">UTF-16</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="toolTip">
          <string>Open databases with foreign keys enabled.</string>
         </property>
         <property name="text">
          <string>&amp;Foreign keys</string>
         </property>
         <property name="buddy">
          <cstring>foreignKeysCheckBox</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="foreignKeysCheckBox">
         <property name="text">
          <string>enabled</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Default &amp;location</string>
         </property>
         <property name="buddy">
          <cstring>locationEdit</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLineEdit" name="locationEdit">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="setLocationButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>&amp;Prefetch block size</string>
         </property>
         <property name="buddy">
          <cstring>spinPrefetchSize</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="spinPrefetchSize">
         <property name="minimum">
          <number>255</number>
         </property>
         <property name="maximum">
          <number>1000000</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Remove line breaks in schema view</string>
         </property>
         <property name="buddy">
          <cstring>checkHideSchemaLinebreaks</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="checkHideSchemaLinebreaks"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>&amp;SQL</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QTreeWidget" name="treeSyntaxHighlighting">
         <property name="showDropIndicator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <column>
          <property name="text">
           <string>Settings name</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Context</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Colour</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Bold</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Italic</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Underline</string>
          </property>
         </column>
         <item>
          <property name="text">
           <string notr="true">keyword</string>
          </property>
          <property name="text">
           <string>Keyword</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string>function</string>
          </property>
          <property name="text">
           <string>Function</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">table</string>
          </property>
          <property name="text">
           <string>Table</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">comment</string>
          </property>
          <property name="text">
           <string>Comment</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">identifier</string>
          </property>
          <property name="text">
           <string>Identifier</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">string</string>
          </property>
          <property name="text">
           <string>String</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
         </item>
         <item>
          <property name="text">
           <string>currentline</string>
          </property>
          <property name="text">
           <string>Current line</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>null</string>
          </property>
          <property name="text">
           <string>Null</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>SQL &amp;editor font size</string>
           </property>
           <property name="buddy">
            <cstring>spinEditorFontSize</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinEditorFontSize">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinLogFontSize">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>SQL &amp;log font size</string>
           </property>
           <property name="buddy">
            <cstring>spinLogFontSize</cstring>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>&amp;Extensions</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Select extensions to load for every database:</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QListWidget" name="listExtensions">
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
           <property name="showDropIndicator" stdset="0">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QToolButton" name="buttonAddExtension">
             <property name="toolTip">
              <string>Add extension</string>
             </property>
             <property name="icon">
              <iconset resource="icons/icons.qrc">
               <normaloff>:/icons/load_extension</normaloff>:/icons/load_extension</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="buttonRemoveExtension">
             <property name="toolTip">
              <string>Remove extension</string>
             </property>
             <property name="icon">
              <iconset resource="icons/icons.qrc">
               <normaloff>:/icons/remove_extension</normaloff>:/icons/remove_extension</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>encodingComboBox</tabstop>
  <tabstop>foreignKeysCheckBox</tabstop>
  <tabstop>locationEdit</tabstop>
  <tabstop>setLocationButton</tabstop>
  <tabstop>checkHideSchemaLinebreaks</tabstop>
  <tabstop>spinPrefetchSize</tabstop>
  <tabstop>treeSyntaxHighlighting</tabstop>
  <tabstop>spinEditorFontSize</tabstop>
  <tabstop>spinLogFontSize</tabstop>
  <tabstop>listExtensions</tabstop>
  <tabstop>buttonAddExtension</tabstop>
  <tabstop>buttonRemoveExtension</tabstop>
 </tabstops>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>saveSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setLocationButton</sender>
   <signal>clicked()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>chooseLocation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>459</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeSyntaxHighlighting</sender>
   <signal>itemDoubleClicked(QTreeWidgetItem*,int)</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>showColourDialog(QTreeWidgetItem*,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonAddExtension</sender>
   <signal>clicked()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>addExtension()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonRemoveExtension</sender>
   <signal>clicked()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>removeExtension()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveSettings()</slot>
  <slot>chooseLocation()</slot>
  <slot>showColourDialog(QTreeWidgetItem*,int)</slot>
  <slot>addExtension()</slot>
  <slot>removeExtension()</slot>
 </slots>
</ui>
